plugins/database/mysql/connection_producer.go

```diff
@@ -13,14 +13,30 @@ import (
 	"sync"
 	"time"
 
+	cloudmysql "cloud.google.com/go/cloudsqlconn/mysql/mysql"
+
 	"github.com/go-sql-driver/mysql"
 	"github.com/hashicorp/go-secure-stdlib/parseutil"
 	"github.com/hashicorp/go-uuid"
+	"github.com/mitchellh/mapstructure"
+
 	"github.com/hashicorp/vault/sdk/database/helper/connutil"
 	"github.com/hashicorp/vault/sdk/database/helper/dbutil"
-	"github.com/mitchellh/mapstructure"
 )
 
+const (
+	authTypeIAM   = "iam"
```

Comment:
should this be "gcp-iam" like the other `authTypeIAM` constant?

Reply:
We are already importing `github.com/hashicorp/vault/sdk/database/helper/connutil` in this file. Can we just reference the constant defined there?

Reply:
We should definitely be careful about what we export from the SDK. But I think this constant is ok. We would want the sdk to be imported and not the other way around which I think you already did 👍